editors/turbo/Portfile

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           github 1.0
+
+github.setup        magiblot turbo 8d42224c563865b232727e93169c81f11cd53fe1
+version             2024.01.07
+revision            0
+categories          editors sysutils
+license             MIT
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         Experimental text editor based on Scintilla and Turbo Vision
+long_description    Turbo is an experimental text editor for the terminal, \
+                    based on the Scintilla code editing component by Neil Hodgson \
+                    and the Turbo Vision application framework.
+
+fetch.type          git
+
+if {${os.platform} eq "darwin" && ${os.major} < 14} {
+    # Lion+ (with Xcode 4.1+) have git; earlier need to bring their own.
+    # However Lion through Mavericks fail with SSL errors.
+    depends_build-append \
+                    port:git
+    git.cmd         ${prefix}/bin/git
+}
+
+post-fetch {
+    system -W ${worksrcpath} "git submodule update --init"
+}
+
+depends_lib-append  port:libmagic \
+                    port:ncurses
+
+compiler.cxx_standard 2017
+
+configure.args-append \
+                    -DTURBO_BUILD_APP=ON \
+                    -DTURBO_BUILD_EXAMPLES=OFF \
+                    -DTURBO_BUILD_TESTS=OFF \
+                    -DTURBO_OPTIMIZE_BUILD=OFF
+
+variant tests description "Build and run tests" {
+    depends_build-append \
+                    port:gtest
+    configure.args-replace \
+                    -DTURBO_BUILD_TESTS=OFF -DTURBO_BUILD_TESTS=ON
+    test.run        yes
+    # Tests will run at the end of the build.
+    # No special target needed.
+    test            { }
```

Comment:
this line is unusual… what issue is it fixing?

Reply:
Otherwise there will be an attempt to run make test, but there is no separate target for that, so we get an error (nothing to do for target, as I recall). It is done it a way that passing a configure arg to CMake builds everything and runs tests. So in Macports terms it is still the build phase.

But we a user will expect to run port test, not just port build.

If there is a better idea, I am open to suggestions.

Reply:
i’ll take a look and see what happens.

Reply:
I see what you mean. It's a bit odd how that works. Rather than override the test phase completely with this:
```
test {}
```
which seems pretty strange when you look at it, while running tests, I tried this, clearing the test command:
```
test.cmd
```

and that worked too. Perhaps that might be a bit more clear to folks what is happening.

I did not see an easy way to separate the build and the test steps in the CMakeLists.txt file, although no doubt it could be done properly with patching. That is a project for upstream, however.

Reply:
Thanks, I will change it to `test.cmd` then.

Reply:
@kencu UPD. Okay, I tried now, it does not behave the same. With `test.cmd` being empty, I get this:
```
[100%] Built target turbo-run-test
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build'
/opt/local/bin/cmake -E cmake_progress_start /opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build/CMakeFiles 0
make: Leaving directory `/opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build'
--->  Testing turbo
Executing:  cd "/opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build" && test 
Command failed:  cd "/opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build" && test 
Exit code: 1
Error: Failed to test turbo: command execution failed
Error: See /opt/local/var/macports/logs/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there is a
bug.
Error: Processing of port turbo failed
```
With `test { }` I get no error:
```
[100%] Built target turbo-run-test
make[2]: Leaving directory `/opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build'
[100%] Built target turbo
make[1]: Leaving directory `/opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build'
/opt/local/bin/cmake -E cmake_progress_start /opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build/CMakeFiles 0
make: Leaving directory `/opt/local/var/macports/build/_opt_svacchanda_SonomaPorts_editors_turbo/turbo/work/build'
--->  Testing turbo
```

Reply:
I think you must have made a typo.

Let me make the suggestion in the code -- I'm supposed to do that more than I do.

Reply:
what you can do is this:
```
    test.cmd        /usr/bin/true
    test.target
```
but as that is arguably no clearer than just disabling the phase altogether, you can just leave it as you did it I think. (You might expect others to comment, more than likely, after it's committed with ideas.)